### ISSUE TITLE
updating state

### ISSUE THREAD

**Issue by forresto** [engraftdev/engraft#9]
What's the engraft equivalent of `const [query, setQuery] = useState('');`?

Trying to find how to update "state" via html `<input>`. The [Synonimizer example](https://github.com/engraftdev/engraft/blob/49ba9d1fb3928320fdaf228f0509704c4f510672/packages/use-engraft-demo-js/src/App.jsx#L42) is doing it outside of engraft. 

Rubberducking here, in case it helps. 🦆 

![image](https://github.com/engraftdev/engraft/assets/395307/1212896f-5efe-49b9-b514-281bbe3b7c9d)
...
![image](https://github.com/engraftdev/engraft/assets/395307/a19a8c1c-44c5-4a3e-9c46-a204c0240a34)

Garden link: https://engraft.dev/graft-garden/#/kBC5VsxmNRsvZaoksDnb/view

**Comment by forresto** [comment]
After digging in, I found this pattern that works...
``` jsx
  <label>
    Search: <input value={state} onChange={
      (e) => {
        // Undocumented
        stateUP.$set(e.target.value);
      }
    } />
  </label>
```

**Comment by joshuahhh** [comment]
That's right. Engraft itself is stateless – an Engraft program acts as a pure function. Any state must come from a host embedding, whether that's Graft Garden (which you show here) or a web-app using useEngraft or something else.
